tests/test_salesforce_incremental_table_reset.py

```diff
@@ -0,0 +1,141 @@
+        streams_replication_methods = {stream: self.INCREMENTAL
+                                       for stream in expected_streams}
+        self.set_replication_methods(conn_id, catalog_entries, streams_replication_methods)
+
+        # Run a sync job using orchestrator
+        first_sync_record_count = self.run_and_verify_sync(conn_id)
+        first_sync_records = runner.get_records_from_target_output()
+
+        first_sync_bookmarks = menagerie.get_state(conn_id)
+
+        # UPDATE STATE for Table Reset
+        new_states = {'bookmarks': dict()}
+        stream_to_current_state = {stream : bookmark.get(self.expected_replication_keys()[stream].pop())
+                                   for stream, bookmark in first_sync_bookmarks['bookmarks'].items()}
+        for stream in stream_to_current_state:
+            replication_key = list(replication_keys[stream])[0]
+            # Remove stream User to reset
+            if stream != 'User':
+                new_states['bookmarks'][stream] = {replication_key: stream_to_current_state[stream]}
+        menagerie.set_state(conn_id, new_states)
+
+        # SYNC 2
+        second_sync_record_count = self.run_and_verify_sync(conn_id)
+        second_sync_records = runner.get_records_from_target_output()
+        second_sync_bookmarks = menagerie.get_state(conn_id)
+
+        # Test by stream
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+                # record counts
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+
+                # data from record messages
+                first_sync_messages = [record.get('data') for record in
+                                       first_sync_records.get(stream).get('messages')
+                                       if record.get('action') == 'upsert']
+                second_sync_messages = [record.get('data') for record in
+                                        second_sync_records.get(stream).get('messages')
+                                        if record.get('action') == 'upsert']
+
+                # replication key for comparing data
+                self.assertEqual(1, len(list(replication_keys[stream])),
+                                 msg="Compound primary keys require a change to test expectations")
+                replication_key = list(replication_keys[stream])[0]
+
+                # bookmarked states (top level objects)
+                first_bookmark_key_value = first_sync_bookmarks.get('bookmarks').get(stream)
+                second_bookmark_key_value = second_sync_bookmarks.get('bookmarks').get(stream)
+
+                # bookmarked states (actual values)
+                first_bookmark_value = first_bookmark_key_value.get(replication_key)
+                second_bookmark_value = second_bookmark_key_value.get(replication_key)
+
+                filtered_synced_records = [
```

Comment:
I'd like to include which sync is being filtered in the variable name for readability.  Maybe filtered_second_sync_records?

Reply:
changed the variable name to filtered_second_sync_records for better readability